httpbp/errors.go

```diff
@@ -0,0 +1,621 @@
+// after fetching a new one.
+func Unauthorized() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusUnauthorized,
+		"UNAUTHORIZED",
+		"This server could not verify that you are authorized to access the document you requested.",
+	)
+}
+
+// PaymentRequired is for 402 responses.
+//
+// PaymentRequired is reserved for future use but has no standard around its
+// use. It is intended to communicate to the client that their request can not
+// be completed until a payment is made.
+func PaymentRequired() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusPaymentRequired,
+		"PAYMENT_REQUIRED",
+		"This server cannot process the request until payment is made.",
+	)
+}
+
+// Forbidden is for 403 responses.
+//
+// This is appropriate for when you can authenticate a request but the client
+// does not have access to the requested resource.
+//
+// Unlike Unauthorized, refreshing an authentication resource and trying again
+// will not make a difference.
+func Forbidden() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusForbidden,
+		"FORBIDDEN",
+		"You do not have access to the requested resource.",
+	)
+}
+
+// NotFound is for 404 responses.
+//
+// This is appropriate for when the client tries to access something that does
+// not exist.
+func NotFound() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusNotFound,
+		"NOT_FOUND",
+		"The requested resource could not be found.",
+	)
+}
+
+// Conflict is for 409 responses.
+//
+// This is appropriate for when a client request would cause a conflict with
+// the current state of the server.
+func Conflict() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusConflict,
+		"CONFLICT",
+		"This request has a conflict with the current state of the system.",
+	)
+}
+
+// Gone is for 410 responses.
+//
+// This is appropriate for when the resource requested was once available but is
+// no longer.
+func Gone() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusGone,
+		"GONE",
+		"The requested resource is no longer available.",
+	)
+}
+
+// PayloadTooLarge is for 413 responses.
+//
+// This is appropriate for when the client sends a request that is larger than
+// the limits set by the server, such as when they try to upload a file that is
+// too big.
+func PayloadTooLarge() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusRequestEntityTooLarge,
+		"PAYLOAD_TOO_LARGE",
+		"The request was too large.",
+	)
+}
+
+// UnsupportedMediaType is for 415 responses.
+//
+// This is appropriate for when the request has an unsupported media format.
+func UnsupportedMediaType() *ErrorResponse {
+	return NewErrorResponse(
+		http.StatusUnsupportedMediaType,
+		"UNSUPPORTED_MEDIA_TYPE",
+		"The given media type is not supported.",
+	)
+}
+
+// Teapot is for 418 responses.
+//
+// This is appropriate for when the server is a teapot rather than a coffee maker.
+func Teapot() *ErrorResponse {
```

Comment:
I thought this has been deprecated but there has been the [Save 418 Movement](https://en.wikipedia.org/wiki/Hyper_Text_Coffee_Pot_Control_Protocol#Save_418_movement)

Glad we're taking a stance here.